backend/app/main.py

```diff
@@ -407,10 +428,184 @@ def update_match(
     )
     if access:
         if crud.get_match(db, match_id=match_id) is None:
-            raise HTTPException(status_code=404, detail="Player not found")
+            raise HTTPException(status_code=404, detail="Match not found")
         crud.update_match(db, match_id=match_id, match=match)
     else:
         raise HTTPException(
             status_code=404, detail="Permission denied, requires at least: moderator"
         )
 
+# tournaments
+@app.post("/api/tournaments/", response_model=schemas.Tournament)
+def create_tournament(
+    tournament: schemas.TournamentCreate,
+    firebase_id: str = Header(None),
+    db: Session = Depends(get_db),
+):
+    access = permissions.is_accessible(
+        db=db, firebase_id=firebase_id, clearance="moderator"
+    )
+    if access:
+        if tournament.tournament_type not in ["round-robin"]:
+            raise HTTPException(status_code=404, detail="Tournament type unknown")
+        teams_ids = tournament.teams_ids
+        for team_id in teams_ids:
+            check = crud.get_team(db, team_id)
+            if check is None:
+                raise HTTPException(
+                    status_code=404, detail="Team " + str(team_id) + " not found"
+                )
+
+        return crud.create_tournament(db=db, tournament=tournament)
+    else:
+        raise HTTPException(
+            status_code=404, detail="Permission denied, requires at least: moderator"
+        )
+
+@app.get("/api/tournaments/", response_model=List[schemas.Tournament])
+def read_tournaments(
+    firebase_id: str = Header(None),
+    skip: int = 0,
+    limit: int = 100,
+    db: Session = Depends(get_db),
+):
+    access = permissions.is_accessible(
+        db=db, firebase_id=firebase_id, clearance="player"
+    )
+    if access:
+        tournaments = crud.get_tournaments(db, skip=skip, limit=limit)
+        return tournaments
+    else:
+        raise HTTPException(
+            status_code=404, detail="Permission denied, requires at least: player"
+        )
+
+@app.get("/api/tournaments/{tournament_id}", response_model=schemas.Tournament)
+def read_tournament(
+    tournament_id: int = Header(None),
+    firebase_id: str = Header(None),
+    db: Session = Depends(get_db),
+):
+    access = permissions.is_accessible(
+        db=db, firebase_id=firebase_id, clearance="player"
+    )
+    if access:
+        db_tournament = crud.get_tournament(db, tournament_id=tournament_id)
+        if db_tournament is None:
+            raise HTTPException(status_code=404, detail="Tournament not found")
+        return db_tournament
+    else:
+        raise HTTPException(
+            status_code=404, detail="Permission denied, requires at least: player"
```

Comment:
`401` / `403`. Looks like this access checking can be an utility function, its used in many places